src/navigators/TabNavigator.tsx

```diff
@@ -69,54 +72,83 @@ const renderSimpleLineIcon = (
 /**
  * Handles tab navigation for the main section of the app
  */
-export const TabNavigator = () => (
-  <Tab.Navigator
-    initialRouteName={ROUTES.WAIT_TIMES}
-    screenOptions={{
-      tabBarLabelStyle: styles.tabBarLabels,
-      headerTitleAlign: "center",
-    }}
-  >
-    <Tab.Screen
-      name={ROUTES.WAIT_TIMES}
-      component={WaitTimesNavigator}
-      options={{
-        title: "Wait Times",
-        headerShown: false,
-        tabBarIcon: ({ color, size }) =>
-          renderFeatherIcon("clock", color, size),
-      }}
-    />
-    <Tab.Screen
-      name={ROUTES.MAP}
-      component={MapScreen}
-      options={{
-        title: "Map",
-        headerShown: false,
-        tabBarIcon: ({ color, size }) =>
-          renderFeatherIcon("map-pin", color, size),
-      }}
-    />
-    <Tab.Screen
-      name={ROUTES.REWARDS}
-      component={RewardsScreen}
-      options={{
-        title: "Rewards History",
-        tabBarLabel: "Rewards",
-        tabBarIcon: ({ color, size }) => renderFeatherIcon("gift", color, size),
-      }}
-    />
-    <Tab.Screen
-      name={ROUTES.PROFILE}
-      component={ProfileStackNavigator}
-      options={{
-        headerShown: false,
-        tabBarIcon: ({ color, size }) =>
-          renderSimpleLineIcon("user", color, size),
+export const TabNavigator = () => {
+  const appState = useRef(AppState.currentState);
+  const [_, setAppStateVisible] = useState(appState.current);
+
+  // Checking if the app is in foreground or background and triggering the fetch of background location permission each time the app is brought to foreground
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", (nextAppState) => {
+      if (
+        appState.current.match(/inactive|background/) &&
+        nextAppState === "active"
+      ) {
+        Location.getBackgroundPermissionsAsync().then((response) => {
+          if (!response.granted) {
+            requestPermissions();
+          }
+        });
+      }
+
+      appState.current = nextAppState;
+      setAppStateVisible(appState.current);
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  });
+
```

Comment:
I think this would be refactored nicely into a custom hook under `/hooks`. That way we can just call it here and keep the navigator clean. What do you think?